### PR TITLE
On some platforms __ARM_NEON is enough to decide we have Neon support

### PR DESCRIPTION
**Pull request by serge-sans-paille**

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
@JohanMabille I'm testing that one, please don't merge before I give you a green light.

File: include/xsimd/config/xsimd_config.hpp [hunk 1]
```diff
@@ -251,7 +251,7 @@
  *
```
Comment (ricotz) [inline include/xsimd/config/xsimd_config.hpp:251]: I think the problem it located above, so when "#ifdef __ARM_NEON" is not enough.

Comment (serge-sans-paille) [reply include/xsimd/config/xsimd_config.hpp:251]: oh, I missed that indeed

Comment (serge-sans-paille) [reply include/xsimd/config/xsimd_config.hpp:251]: @ricotz maybe better like this?

Comment (ricotz) [reply include/xsimd/config/xsimd_config.hpp:251]: I guess the problem is on our side and needs to be handled downstream.
So if there is no actual problem then please hold back those changes.

Comment (serge-sans-paille) [reply include/xsimd/config/xsimd_config.hpp:251]: ack, please ping me whenever you find an out.

Comment (ricotz) [reply include/xsimd/config/xsimd_config.hpp:251]: Thank for looking into it, but this is change is not reasonable for upstream.
It is fixed downstream by respecting specifics of Ubuntu's toolchain.

**Comment by serge-sans-paille** [comment]
Looks like the original issue was on mozilla side, I do think this patch is not actually needed, see https://bugzilla.mozilla.org/show_bug.cgi?id=1822901

**Pull request closed without merging**